ee/clickhouse/views/insights.py

```diff
@@ -8,17 +8,21 @@
 from ee.clickhouse.queries.funnels.funnel_correlation import FunnelCorrelation
 from posthog.api.insight import InsightViewSet
 from posthog.decorators import cached_function
-from posthog.models import User
+from posthog.models import Insight, User
 from posthog.models.filters import Filter
 
 
 class CanEditInsight(BasePermission):
     message = "This insight is on a dashboard that can only be edited by its owner, team members invited to editing the dashboard, and project admins."
 
-    def has_object_permission(self, request: Request, view, insight) -> bool:
-        if request.method in SAFE_METHODS or insight.dashboard_id is None:
+    def has_object_permission(self, request: Request, view, insight: Insight) -> bool:
+        if request.method in SAFE_METHODS:
             return True
-        return insight.dashboard.can_user_edit(cast(User, request.user).id)
+        dashboards = list(insight.dashboards.all())
+        if not dashboards:
+            return True
+        edit_permissions = [d.can_user_edit(cast(User, request.user).id) for d in dashboards if d is not None]
+        return any(edit_permissions)
```

Comment:
@clarkus one UI thing that will change here is that, once Insights can be added to multiple dashboards (and we're not adding that in this PR), an insight could be on a private dashboard *and* on an editable dashboard. 

Do we need to communicate this when viewing permissions?

presumably text in the popover here

![Uploading Screenshot 2022-04-20 at 13.35.39.png…]()

Reply:
Can users edit the insight definitions, or just the "references" on the dashboard? As an example, does edit in this context mean insight card size, position, placement, etc? Or does it mean I can use the dashboard as a means of editing an insight definition I wouldn't otherwise have access to edit? Said another way, could an insight referenced on multiple dashboards and positioned / sized differently on each dashboard?